resource/cmake/CommonFunctions.cmake

```diff
@@ -0,0 +1,443 @@
+    set(MULTI_VALUE_ARGS DEPENDENCIES)
+
+    cmake_parse_arguments(_COMMON_FUNCTIONS
+        "${BOOLEANS}"
+        "${SINGLE_VALUE_ARGS}"
+        "${MULTI_VALUE_ARGS}"
+        ${ARGN}
+    )
+
+    if (NOT SPHINX_BIN)
+        set(SPHINX_BIN "sphinx-build")
+    endif()
+
+    add_custom_command(
+        OUTPUT
+            ${_COMMON_FUNCTIONS_OUTPUT_DIR}
+        COMMENT
+            "Generating ${_COMMON_FUNCTIONS_PLUGIN_NAME} Documentation"
+        COMMAND
+            ${SPHINX_BIN}
+                ${_COMMON_FUNCTIONS_INPUT_DIR}
+                ${_COMMON_FUNCTIONS_OUTPUT_DIR}
+                ${SPHINX_OPTS}
+        DEPENDS
+            ${_COMMON_FUNCTIONS_DEPENDENCIES}
+        VERBATIM
+    )
+
+    add_custom_target(${_COMMON_FUNCTIONS_CUSTOM_TARGET_NAME} ALL
+        DEPENDS
+            "${_COMMON_FUNCTIONS_OUTPUT_DIR}"
+    )
+
+    if (${_COMMON_FUNCTIONS_DOXYGEN_ENABLED})
+        if (NOT DOXYGEN_BIN)
+            set(DOXYGEN_BIN "doxygen")
+        endif()
+
+        add_custom_command(
+            OUTPUT
+                "${_COMMON_FUNCTIONS_INPUT_DIR}/doxyoutput"
+            COMMAND
+                ${DOXYGEN_BIN} "${_COMMON_FUNCTIONS_INPUT_DIR}/Doxyfile"
+            WORKING_DIRECTORY
+                "${_COMMON_FUNCTIONS_INPUT_DIR}"
+            DEPENDS
+                "${_COMMON_FUNCTIONS_INPUT_DIR}/Doxyfile"
+            VERBATIM
+        )
+        add_custom_target("${_COMMON_FUNCTIONS_CUSTOM_TARGET_NAME}_DOXYGEN"
+            DEPENDS
+                "${_COMMON_FUNCTIONS_INPUT_DIR}/doxyoutput"
+        )
+
+        add_dependencies(${_COMMON_FUNCTIONS_CUSTOM_TARGET_NAME}
+            ${_COMMON_FUNCTIONS_CUSTOM_TARGET_NAME}_DOXYGEN)
+    endif()
+
+endfunction()
+
+macro(m_configure_plugin_defines)
+    if (RTIGATEWAY_ENABLE_LOG
+            OR CMAKE_BUILD_TYPE STREQUAL "Debug"
+            OR NOT CMAKE_BUILD_TYPE)
+        list(APPEND ${RSPLUGIN_PREFIX}_DEFINES ${RSPLUGIN_PREFIX}_ENABLE_LOG)
+    else()
+        list(APPEND ${RSPLUGIN_PREFIX}_DEFINES ${RSPLUGIN_PREFIX}_DISABLE_LOG)
+    endif()
+
+    if (RTIGATEWAY_ENABLE_TRACE)
+        list(APPEND ${RSPLUGIN_PREFIX}_DEFINES ${RSPLUGIN_PREFIX}_ENABLE_TRACE)
+    endif()
+
+    if (RTIGATEWAY_ENABLE_SSL)
+        list(APPEND ${RSPLUGIN_PREFIX}_DEFINES ${RSPLUGIN_PREFIX}_ENABLE_SSL)
+    endif()
+
+    set(${RSPLUGIN_PREFIX}_DEFINES     "${${RSPLUGIN_PREFIX}_DEFINES}"
+            CACHE INTERNAL "List compiler defines")
+endmacro()
+
+function(set_required_variable _OUT_VARIABLE)
+    set(candidate_vars ${ARGN})
+    foreach(v ${candidate_vars})
+        # Give preference to variables passed directly to cmake
+        if (DEFINED ${v})
+            message(STATUS "${_OUT_VARIABLE} = ${${v}}")
+            set(${_OUT_VARIABLE} ${${v}} PARENT_SCOPE)
+            return()
+        elseif (DEFINED ENV{${v}})
+            message(STATUS "${_OUT_VARIABLE} = $ENV{${v}}")
+            set(${_OUT_VARIABLE} $ENV{${v}} PARENT_SCOPE)
```

Comment:
I haven't had issues when using environment variables with Windows-style paths. In which situations can this be a problem?

Reply:
I have always had problem reading NDDSHOME/CONNEXTDDS_DIR from the environment on WIndows. I doubt it that it's been "fixed", since I most recently run into it while developing the RMW.

Reply:
If things are working for you, I'm fine not making changes for now.

Reply:
ok, resolving this thread